File: Arrays&Hashing/contains_duplicate/contains-duplicate.ts
function hasDuplicate(nums: number[]): boolean {
    const numbers = new Set<number>();
    for (const num of nums) {
        if (numbers.has(num)) {
            return true;
        }
        numbers.add(num);
    }
    return false;
}

const test1: number[] = [1, 2, 3, 1];
const test2: number[] = [1, 2, 3, 4];
const test3: number[] = [1, 1, 1, 3, 3, 4, 3, 2, 4, 2];

console.log("Test 1:", hasDuplicate(test1));
console.log("Test 2:", hasDuplicate(test2));
console.log("Test 3:", hasDuplicate(test3));

File: src/main/java/arrays_hashing/ContainsDuplicate.java
package arrays_hashing;

import java.util.Set;
import java.util.HashSet;

public class ContainsDuplicate {
    public boolean hasDuplicate(int[] nums) {
        Set<Integer> seenNumbers = new HashSet<>();
        for (int num: nums) {
            if (seenNumbers.contains(num))
                return true;
            seenNumbers.add(num);
        }
        return false;
    }

    public static void main(String[] args) {
        ContainsDuplicate solution = new ContainsDuplicate();

        int[] test1 = {1,2,3,1};
        int[] test2 = {1,2,3,4};
        int[] test3 = {1,1,1,3,3,4,3,2,4,2};

        System.out.println("Test 1: " + solution.hasDuplicate(test1));
        System.out.println("Test 2: " + solution.hasDuplicate(test2));
        System.out.println("Test 3: " + solution.hasDuplicate(test3));
    }
}